src/transformers/modeling_utils.py

```diff
@@ -3840,7 +3840,6 @@ def _fix_key(key):
             model_buffers = {".".join([prefix, key]) for key in model_buffers}
         unexpected_keys = sorted(unexpected_keys - model_buffers)
 
-        model.tie_weights()
```

Comment:
prob we can remove this, given a few lines above the weights before loading state dict are tied already

Reply:
This was introduced in #25107, however, indeed there are no changeds to the model in the code above. 
Can you make sure the bug fixed in the PR are still passing?